packages/components/src/local/mapping/helpers/createGrid.ts

```diff
@@ -50,17 +51,55 @@ const createGrid = (bounds: L.LatLngBounds, tileDiameterMins: number): SergeGrid
     const newCell: SergeHex<{}> = cell as SergeHex<{}>
     newCell.name = cellName(newCell)
 
-    // generate the cenll centre in Leaflet coords
+    // generate the cell centre in Leaflet coords
     const centreHex: Point = cell.toPoint()
     newCell.centreLatLng = toWorld(centreHex, correctedOrigin, tileSizeDegs / 2)
     return newCell
   })
   const unTyped: any = newCells
   const asSerge: Grid<SergeHex<{}>> = unTyped as Grid<SergeHex<{}>>
 
+  // sort out corner offset
+  // the hexes all have the same corners object, so just use the first one
+  const hexOne = asSerge[0]
+
+  // get the coordinates of the centre of the hex, relative
+  // to the top-left origin
+  const centreH = hexOne.center()
+
+  // and the coords of the top-left origin
+  const cellOrigin = hexOne.coordinates()
+
+  // capture the offset between a cell centre, and the cell origin
+  const centreOffset = L.point(centreH).subtract(L.point(cellOrigin))
+  
   const sergeGrid: SergeGrid<SergeHex<{}>> = asSerge as SergeGrid<SergeHex<{}>>
   sergeGrid.origin = correctedOrigin
   sergeGrid.tileDiameterDegs = tileSizeDegs
+  sergeGrid.centerOffset = centreOffset
+  sergeGrid.toScreen = (point: L.LatLng): PointLike => {
```

Comment:
I don't really understand what is happening here. Are you assigning a function to a variable in `sergeGrid`? If so, why? You already have a helper called `toScreen` so what is going on here, can't you just pass data to that? Can you add comments so it's clear?

The same goes for the other instances of `sergeGrid.thing` below.

Reply:
Sure, making them class members, they can refer to other class data (particularly the origin & grid size).  That means that in the code, when we have a `gridCells` object, we can just call `gridCells.toWorld('H02')` - and it will us the origin/tilesize that's already stored in the grid.

Reply:
It's a way of hiding the value inside the method.  The `toWorld` method does not need the cell origin or grid size passed to it, since it can access it from the parent object.  It also reduces the need for other code to know about how to do coordinate transforms.

Reply:
Fair enough, can you add some comments to the code to make it clear for the future.